src/tests/user.test.js

```diff
@@ -443,164 +443,239 @@ describe('POST /api/v1/users', () => {
+        done();
+      });
+  });
+});
+describe('POST /api/v1/reset-password/:token', () => {
+  it('Should return error if user tries to reset password with Invalid confirmin password', (done) => {
+    chai.request(app)
+      .post(`/api/v1/reset-password/${userToken}`)
+      .send(dummyUser.numresetConfirmPassword)
+      .end((err, res) => {
+        if (err) done(err);
+        expect(res).have.status(400);
+        expect(res).to.be.an('object');
+        expect(res.body).to.have.keys('status', 'error');
+        expect(res.body.error).to.deep.equal('Confirm password must be a string');
+        expect(res.body.status).to.deep.equal('failed');
+        done();
+      });
+  });
+});
+// Login Tests
+describe('POST /api/v1/login', () => {
+  it('Should output error if user provides no email', () => {
+    chai
+      .request(app)
+      .post('/api/v1/login')
+      .send({
+        email: '',
+        password: 'Jamal1230!'
+      })
+      .end((err, res) => {
+        expect(res).have.status(400);
+        expect(res.body).to.have.key('error');
+        expect(res.body.error).to.deep.equal('email is not allowed to be empty');
+      });
+  });
+  it('Should output error if user provides no password', () => {
+    chai
+      .request(app)
+      .post('/api/v1/login')
+      .send({
+        email: 'jamal@gmail12.com',
+        password: ''
+      })
+      .end((err, res) => {
+        expect(res).have.status(400);
+        expect(res.body).to.have.key('error');
+        expect(res.body.error).to.deep.equal('password is not allowed to be empty');
+      });
+  });
+
+  it('Should output error message when password provided is incorrect', () => {
+    chai
+      .request(app)
+      .post('/api/v1/login')
+      .send({
+        email: 'jamal@gmail12.com',
+        password: 'Jamal12301'
+      })
+      .end((err, res) => {
+        expect(res).have.status(401);
+        expect(res.body).to.have.key('error');
+        expect(res.body.error).to.deep.equal('Incorrect email or password');
+      });
+  });
+
+  it('Should output error message when email provided is incorrect', () => {
+    chai
+      .request(app)
+      .post('/api/v1/login')
+      .send({
+        email: 'jamalmoh@gmail12.com',
+        password: 'Jamal1230'
+      })
+      .end((err, res) => {
+        expect(res).have.status(401);
+        expect(res.body).to.have.key('error');
+        expect(res.body.error).to.deep.equal('Incorrect email or password');
+      });
+  });
+  it('Should return error message when user account is not verified', () => {
+    chai
+      .request(app)
+      .post('/api/v1/login')
+      .send({
+        email: 'hareraloston@gmail.com',
+        password: 'Jamal.123',
+      })
+      .end((err, res) => {
+        expect(res).have.status(401);
+        expect(res.body).to.have.key('error');
+        expect(res.body.error).to.deep.equal('Please verify your account first. Visit your email to verify');
+      });
   });
 });
```

Comment:
In the tests here. I don't see the test for a successful login. Why didn't you test for a successful login?

Reply:
done